File: schemas/digital-asset-management/drizzle/download_presets.ts
// download_presets: Configurable output format presets for asset downloads.
// See README.md for full design rationale.

import { pgTable, uuid, text, integer, timestamp, index } from "drizzle-orm/pg-core";
import { users } from "../../auth-rbac/drizzle/users";
import { workspaces } from "./workspaces";

export const downloadPresets = pgTable(
  "download_presets",
  {
    id: uuid("id").primaryKey().defaultRandom(),
    workspaceId: uuid("workspace_id").notNull().references(() => workspaces.id, { onDelete: "cascade" }),
    name: text("name").notNull(),
    outputFormat: text("output_format"),
    maxWidth: integer("max_width"),
    maxHeight: integer("max_height"),
    quality: integer("quality"),
    dpi: integer("dpi"),
    createdBy: uuid("created_by").notNull().references(() => users.id, { onDelete: "restrict" }),
    createdAt: timestamp("created_at", { withTimezone: true }).notNull().defaultNow(),
    updatedAt: timestamp("updated_at", { withTimezone: true }).notNull().defaultNow(),
  },
  (table) => [
    index("idx_download_presets_workspace_id").on(table.workspaceId),
  ]
);

File: schemas/digital-asset-management/spacetimedb/download_presets.rs
// download_presets: Predefined output configurations for asset downloads.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

#[spacetimedb::table(name = download_presets, public)]
pub struct DownloadPreset {
    #[primary_key]
    pub id: String, // UUID

    #[index(btree)]
    pub workspace_id: String, // FK → workspaces.id (cascade delete)

    pub name: String,
    pub output_format: Option<String>,
    pub max_width: Option<i32>,
    pub max_height: Option<i32>,
    pub quality: Option<i32>,
    pub dpi: Option<i32>,

    pub created_by: String, // FK → users.id (restrict delete)

    pub created_at: Timestamp,
    pub updated_at: Timestamp,
}
